File: macos/Mindbase/DebugLog.swift
import Foundation

enum DebugLog {
    private static let path = "/Volumes/Lacie/Users/osarogie/code/ubase/.cursor/debug-c2f09c.log"

    static func write(location: String, message: String, hypothesisId: String, data: [String: String] = [:]) {
        var payload: [String: Any] = [
            "sessionId": "c2f09c",
            "timestamp": Int(Date().timeIntervalSince1970 * 1000),
            "location": location,
            "message": message,
            "hypothesisId": hypothesisId,
            "data": data,
        ]
        guard let json = try? JSONSerialization.data(withJSONObject: payload),
              let line = String(data: json, encoding: .utf8) else { return }
        let url = URL(fileURLWithPath: path)
        if !FileManager.default.fileExists(atPath: path) {
            FileManager.default.createFile(atPath: path, contents: nil)
        }
        guard let handle = try? FileHandle(forWritingTo: url) else { return }
        defer { try? handle.close() }
        try? handle.seekToEnd()
        if let data = (line + "\n").data(using: .utf8) {
            try? handle.write(contentsOf: data)
        }
    }
}

File: internal/debuglog/debuglog.go
package debuglog

import (
	"encoding/json"
	"os"
	"time"
)

const Path = "/Volumes/Lacie/Users/osarogie/code/ubase/.cursor/debug-c2f09c.log"

func Write(location, message, hypothesisID string, data map[string]any) {
	payload := map[string]any{
		"sessionId":    "c2f09c",
		"timestamp":    time.Now().UnixMilli(),
		"location":     location,
		"message":      message,
		"hypothesisId": hypothesisID,
		"data":         data,
	}
	line, err := json.Marshal(payload)
	if err != nil {
		return
	}
	f, err := os.OpenFile(Path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
	if err != nil {
		return
	}
	defer f.Close()
	_, _ = f.Write(append(line, '\n'))
}
